File: app/src/main/java/com/wjf/self_demo/MyApp.java
package com.wjf.self_demo;

import android.os.Handler;
import android.os.HandlerThread;
import android.view.Choreographer;

import com.github.moduth.blockcanary.BlockCanary;
import com.wjf.barcode.BeepManager;
import com.wjf.barcode.CustomException;
import com.wjf.self_demo.frame.fps.AppBlockCanaryContext;
import com.wjf.self_demo.frame.fps.FPSFrameCallback2;
import com.wjf.self_demo.frame.fps.HandlerBlockTask;
import com.wjf.self_library.common.BaseApplication;
import com.wjf.self_library.http.HttpManager;

/** @author asus */
public class MyApp extends BaseApplication {

    private static final String SP_TAG = "MyApp";
    private HandlerThread mBlockThread = new HandlerThread("blockThread");

    @Override
    public void onCreate() {
        super.onCreate();
        HttpManager.Builder builder =
                new HttpManager.Builder().timeout(5L).baseUrl(HttpService.baseUrl);
        HttpManager.Instance.init(builder);
        BeepManager.init(this, SP_TAG);
/*        BlockCanary.install(this, new AppBlockCanaryContext()).start();
        mBlockThread.start();
        Choreographer.getInstance()
                .postFrameCallback(new FPSFrameCallback2(new Handler(mBlockThread.getLooper())));
        new HandlerBlockTask().startWork();*/
        try {
            BeepManager.getInstance()
                    .setBeepRes(R.raw.beep)
                    .setBeepVolume(0.10f)
                    .setVibrateDuration(200L)
                    .build();
        } catch (CustomException e) {
            e.printStackTrace();
        }
    }
}

File: app/src/main/java/com/wjf/self_demo/MyApp.kt
package com.wjf.self_demo

import android.app.Activity
import android.os.Bundle
import android.os.HandlerThread
import org.jxxy.debug.barcode.BeepManager
import org.jxxy.debug.barcode.CustomException
import org.jxxy.debug.corekit.common.BaseApplication
import java.util.*

/** @author asus
 */
class MyApp : BaseApplication() {
    private val mBlockThread = HandlerThread("blockThread")

    val stack = LinkedList<Activity?>() as Deque<Activity?>
    private val mCallbacks: ActivityLifecycleCallbacks = object :
        ActivityLifecycleCallbacks {
        override fun onActivityCreated(activity: Activity, savedInstanceState: Bundle?) {
            stack.push(activity)
        }

        override fun onActivityStarted(activity: Activity) {}
        override fun onActivityResumed(activity: Activity) {}
        override fun onActivityPaused(activity: Activity) {}
        override fun onActivityStopped(activity: Activity) {}
        override fun onActivitySaveInstanceState(activity: Activity, outState: Bundle) {}
        override fun onActivityDestroyed(activity: Activity) {
            stack.pop()
        }
    }

    override fun onCreate() {
        super.onCreate()
        BeepManager.init(this, SP_TAG)
//        IconFontManager.initAsset("iconfont.ttf")
        //        BlockCanary.install(this, new AppBlockCanaryContext()).start();
//        mBlockThread.start()
//        Choreographer.getInstance()
//            .postFrameCallback(
//                FPSFrameCallback2(Handler(mBlockThread.looper), stack)
//            )
        //        new HandlerBlockTask().startWork();

        registerActivityLifecycleCallbacks(mCallbacks)
        try {
            BeepManager.getInstance()
                .setBeepRes(R.raw.beep)
                .setBeepVolume(0.10f)
                .setVibrateDuration(200L)
                .build()
        } catch (e: CustomException) {
            e.printStackTrace()
        }
    }

    override fun httpBaseUrl(): String = "http://112.124.53.30:9001/"

    override fun iconFontPath(): String = "iconfont.ttf"

    companion object {
        private const val SP_TAG = "MyApp"
    }
}
